### PR TITLE
[CORE-1867] Fix pachtl auth login

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/internal/client/client.go [hunk 1]
```diff
@@ -585,8 +585,11 @@ func newOnUserMachine(ctx context.Context, cfg *config.Config, context *config.C
 		client.authenticationToken = context.SessionToken
 	}
 
-	// Verify cluster deployment ID
+	// Verify cluster deployment ID and project.
 	clusterInfo, err := client.InspectClusterWithVersionAndProject(version.Version, &pfs.Project{Name: context.Project})
+	if err != nil && status.Code(err) == codes.Unauthenticated {
```
Comment (albscui) [inline src/internal/client/client.go:590]: Should we also check for `codes.ErrBadToken`? In that case, the use would also need to login.

Comment (robert-uhl) [reply src/internal/client/client.go:590]: There is no `codes.ErrBadToken`; I think that you mean `auth.ErrBadToken`, which is a `codes.Unauthenticated` error: https://github.com/pachyderm/pachyderm/blob/325662980c95803503c419c0c94f100bdd95182d/src/auth/auth.go#L141

Arguably we should also retry when the error is `codes.PermissionDenied`, though. I’ll push that update in a second.

**Review by albscui — APPROVED** [review]

**Review by brycemcanally — APPROVED** [review]
lgtm, just remove the `status.FromError` checks since they are unnecessary.

**Comment by robert-uhl** [comment]
Merging despite failed `govulncheck` because the only available fix is in an unstable version of Go.

**Pull request merged**